src/shiver/models/sample.py

```diff
@@ -0,0 +1,174 @@
+import numpy
+
+# pylint: disable=no-name-in-module
+from mantid.simpleapi import mtd, DeleteWorkspace, SetUB, CreateSingleValuedWorkspace, LoadIsawUB
+from mantid.geometry import OrientedLattice
+from mantid.kernel import Logger
+from mantidqtinterfaces.DGSPlanner.LoadNexusUB import LoadNexusUB
+from mantidqtinterfaces.DGSPlanner.ValidateOL import ValidateUB
+
+logger = Logger("SHIVER")
+
+
+class SampleModel:
+    """Sample model"""
+
+    def __init__(self, name):
+        self.error_callback = None
+        self.name = name
+        self.oriented_lattice = None
+
+    def connect_error_message(self, callback):
+        """Set the callback function for error messages"""
+        self.error_callback = callback
+
+    # def get_matrix_ub(self):
+    #    # if  'CrystalAlign:UBMatrix' in w.getExperimentInfo(0).run().keys()
+    #    matrix = "0.0100,0.0000,0.0000,0.0000,0.0100,0.0000,0.0000,0.0000,0.0100"
+    #    if mtd.doesExist(self.name):
+    #        matrix_key = ""
+    #        for key in mtd[self.name].getExperimentInfo(0).run().keys():
+    #            if "CrystalAlign::UBMatrix" in key:  # CrystalAlign:UBMatrix
+    #                matrix_key = key
+    #                break
+    #        if matrix_key:
+    #            matrix = mtd[self.name].getExperimentInfo(0).run()[matrix_key].value[0]
+    #            return matrix
+    #    return matrix
+
+    def get_lattice_ub(self):
+        """return oriented lattice object from mtd"""
+        if mtd.doesExist(self.name):
+            return mtd[self.name].getExperimentInfo(0).sample().getOrientedLattice()
+        err_msg = f"Workspace {self.name} does not exist\n"
+        logger.error(err_msg)
+        if self.error_callback:
+            self.error_callback(err_msg)
+        return False
+
+    def get_ub_matrix_from_lattice(self, params):
+        """check and return ub matrix using lattice parameters"""
+        # u and v cannot be colinear ; see projections crossproduct of u and v
+        ub_matrix = []
+        if not self.validate_lattice(params):
+            err_msg = "uv and vx arrays need to be non co-linear\n"
+            logger.error(err_msg)
+            if self.error_callback:
+                self.error_callback(err_msg)
+            return ub_matrix
+        try:
+            uvec = numpy.array([params["latt_ux"], params["latt_uy"], params["latt_uz"]])
+            vvec = numpy.array([params["latt_vx"], params["latt_vy"], params["latt_vz"]])
+            oriented_lattice = OrientedLattice(
+                params["latt_a"],
+                params["latt_b"],
+                params["latt_c"],
+                params["latt_alpha"],
+                params["latt_beta"],
+                params["latt_gamma"],
+            )
+            oriented_lattice.setUFromVectors(uvec, vvec)
+            ub_matrix = oriented_lattice.getUB()
+            self.oriented_lattice = oriented_lattice
+            print("ub_matrix", ub_matrix)
```

Comment:
Either remove this print or log to debug

Reply:
The print is removed.